tests/test_schema_verifier.py

```
@@ -243,7 +243,17 @@ def test_unique_items_invalid(self, verifier):
         schema = {"type": "array", "uniqueItems": True}
         result = verifier.verify([1, 2, 2, 3], schema)
         assert result["is_valid"] == False
-    
+
+    def test_unique_items_uncheckable_fails_closed(self, verifier):
+        """If uniqueness cannot be proven, validation must fail closed."""
+        schema = {"type": "array", "uniqueItems": True}
+        result = verifier.verify([{"bad": {1, 2}}, {"bad": {3, 4}}], schema)
+
+        assert result["is_valid"] == False
```

Ruff (0.15.10)

[error] Avoid equality comparisons to `False`; use `not result["is_valid"]:` for false checks

Replace with `not result["is_valid"]`

(E712)